src/com/generic/tracker/WarehouseTracker.java

```diff
@@ -0,0 +1,176 @@
+ * 
+ * @author Seyi Ola
+ *
+ */
+public class WarehouseTracker {
+	
+	private Map<Integer, Warehouse> warehouses; // Stores a collection of warehouses
+											    // mapped by their id 
+	
+	public WarehouseTracker()
+	{
+		// Store the warehouse
+		warehouses = new HashMap<>();	
+	}
+	
+	/**
+	 * /**
+	 * Adds a new warehouse to the warehouse collection.
+	 * If the warehouse already exists, we return false.
+	 * @param mWarehouse warehouse to add.
+	 * @return true if add was successful,
+	 * 		   false if add wasn't.
+	 */
+	public boolean addWarehouse(Warehouse mWarehouse)
+	{
+		boolean added = false;
+		if(warehouses.get(mWarehouse.getWarehouseID()) == null)
+		{
+			warehouses.put(mWarehouse.getWarehouseID(), mWarehouse);
+			added = true;
+		}
+		return added;
+	}
+	
+	/**
+	 * Adds a shipment to a warehouse, if the warehouseID
+	 * does not exist, it returns -1, if
+	 * the shipment could not be added because it's 
+	 * freight receipt has ended, it returns 0,
+	 * and if the shipment was successfully added, it 
+	 * returns 1.
+	 * @param warehouseID the warehouseID
+	 * @param shipment the shipment to add to this warehouse
+	 * @return -1 if ID does not exist
+	 * 		  , 0 if freight receipt is disabled,
+	 * 		  , 1 if successfully added.
+	 * 			
+	 */
+	public int addShipment(int warehouseID, Shipment mShipment)
+	{
+		Warehouse theWarehouse = warehouses.get(warehouseID);
+		
+		int logCode = 0;
+		
+		if (theWarehouse == null)
+		{
+			logCode = -1;
+		}else if (theWarehouse.addShipment(mShipment))
+		{
+			logCode = 1;
+		}
+		
+		return logCode;
+	}
+	
+	/**
+	 * Enables a warehouse receipt freight, if the warehouseID
+	 * does not exist, it returns -1, if
+	 * the freight receipt could not be enabled because it's 
+	 * freight receipt is enabled already, it returns 0,
+	 * and if the warehouse freight is already enabled, it 
+	 * returns 1.
+	 * @param warehouseID the warehouse ID
+	 * @return log number based on result(-1:non-existent warehouse,
+	 * 									   0: already enabled,
+	 * 									   1: successfully added).
+	 */
+	
+	 public int enableFreight(int warehouseID)
+	 {
+		Warehouse theWarehouse = warehouses.get(warehouseID);
+		
+		int logCode = 0;
+		
+		if (theWarehouse == null) 
+		{
+			logCode = -1;
+		}else if (!theWarehouse.receivingFreight()) 
+		{
+			theWarehouse.enableFreight();
+			logCode = 1;
+		}
+		
+		return logCode;
+	 }
+	 
+	/**
+	 * Disables a warehouse receipt, if the warehouseID does not exist, it returns
+	 * -1, if the freight receipt could not be disabled because it's freight receipt
+	 * is disabled already, it returns 0, and if the warehouse freight is already
+	 * disabled, it returns 1.
+	 * @param warehouseID the warehouse ID
+	 * @return log number based on result(-1:non-existent warehouse,
+	 * 									   0: already disabled,
+	 * 									   1: successfully added).
+	 */
+
+	public int endFreight(int warehouseID) {
+		Warehouse theWarehouse = warehouses.get(warehouseID);
```

Comment:
If a warehouse does not exist, it should throw a RunTimeException.
the other can be boolean

Reply:
Returns boolean for now